GoogleCloudPlatform/datacatalog-connectors-bi
# Use site contentUrl to generate the Linked Resources

**- What I did**
1. Changed the Metadata API scrape results in order to inject the `contentUrl` field into `site` objects.
1. Changed the Data Catalog Entry factory to use this new field when generating the `entry.linkedResource` field.

This PR closes #43.

**- How I did it**
1. Added the `MetadataAPIHelper.__add_site_content_url_field()` method and called it at the end of all public methods of the class.
1. Fixed the `DataCatalogEntryFactory.__format_site_content_url()` method to use the site `contentUrl` instead of the name.

I'm also leveraging this PR to add some improvements to the Tableau connector:
1. The authentication workflows are now fully managed by the API Helper classes to keep consistency with other BI connectors.
1. The scrape result logs are now fully managed by the MetadataScraper class to keep consistency with other BI connectors.
1. Updated the unit tests to cover all changes.
1. Updated the REST API Postman Collection.

**- How to verify it**
Run the unit tests and, **preferably, the connector against a Tableau Server with more than one site** to verify the linked resources generated for entries scraped from the default site and the others.

**- Description for the changelog**
Generate Data Catalog Entry linked resource from the site contentUrl field.
